framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -305,6 +306,21 @@
     /** Mapping from an Element to the source Tree of the declaration. */
     private final Map<Element, Tree> elementToTreeCache;
 
+    /** Viewpoint adaptor to perform viewpoint adaptation */
+    protected final ViewpointAdaptor<?> viewpointAdaptor;
```

Comment:
I emailed you the relationship between the two.
I don't think it this code should be in GenericAnnotatedTypeFactory(lower level). According to [documentation](https://checkerframework.org/api/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.html#addCheckedCodeDefaults-org.checkerframework.framework.util.defaults.QualifierDefaults-), GenericATF is flow related. And the viewpoint adaptation is the functionality that every ATF should inherit from the top ATF - AnnotatedTypeFactory.